modules/sw360/sw360-workflow/src/main/java/org/eclipse/sw360/antenna/sw360/SW360MetaDataUpdater.java

```diff
@@ -104,6 +106,16 @@ public boolean isUploadSources() {
     }
 
     public SW360Release uploadAttachments(SW360Release sw360Release, Map<Path, SW360AttachmentType> attachments) {
-        return releaseClientAdapter.uploadAttachments(sw360Release, attachments);
+        AttachmentUploadRequest.Builder builder = AttachmentUploadRequest.builder(sw360Release);
+        for (Map.Entry<Path, SW360AttachmentType> e : attachments.entrySet()) {
+            builder = builder.addAttachment(e.getKey(), e.getValue());
+        }
+
+        AttachmentUploadResult result = releaseClientAdapter.uploadAttachments(builder.build());
+        LOGGER.debug("Result of attachment upload operation: {}", result);
+        if (!result.isSuccess()) {
+            LOGGER.error("Failed to upload attachments: {}", result.failedUploads());
```

Comment:
When the error is due to a duplicate attachment, this error message will be logged many times. And in case of the duplicate attachments it should not really be an error, since the attachment in the end does exist in SW360 with the release. 
While a bit laborious, could for example the exception be checked for the word duplicate or sth similar? because from a users perspective error messages should only be thrown when there is an acutal error. Dupliactes should be a warning at most

Reply:
Not sure whether this is true in all use cases. The local attachment file may be different from the one on the server. If the library does not return error information in this case, the user would expect that it had been replaced on the server, while it had not.

Reply:
Is there a way to determine if it is the same? because i think printing an error message each time might also lead the user to not take the message seriously in case it actually is a different file.

Reply:
SW360 maintains a hash value for attachments that have been uploaded. So by calculating the hash on the local file (we could add an utility method to support this), one can determine whether files have changed.

It would also be possible to treat duplicates not as errors in general. The upload result object could contain a list with duplicates that have been detected. Then the caller can decide how to treat this situation.

I am afraid that the API related to attachments seems to be very restricted in SW360. A once uploaded attachment file cannot be deleted or overwritten. So if you once uploaded a wrong attachment file, you are stuck with it.

Reply:
The above suggestion sounds promising (and probably worth another isuue / pr) 

maybe we can also check out what changes could be done in sw360.

The for now this should be fine and we can find a solution in a separate PR